rocky-raft/src/rocky/raft/server/CandidateLogic.java

```diff
@@ -14,14 +15,14 @@
     }
 
     @Override
-    public Message process(Message message) {
+    public Message process(Message message, ServerContext serverContext) {
```

Comment:
Putting all server variables to ServerContext is a great idea! This way we can finely control access to them (like synchronization etc). How about we pass on ServerContext in the constructor of CandidateLogic (and other ServerLogic). This way we can have access to the ServerContext from anywhere in the ServerLogic (and not just process method).